### DISCUSSION TITLE
[V8] Row selection with controlled pagination

### DISCUSSION THREAD

**Discussion by dufia** [TanStack/table#4943] [Q&A]
How would you do row selection with controlled, async pagination? Basically, I make a selection, go to another page (load new data, get rid of previous data), navigate to the previous page and still see previously selected rows selected?

**Comment by lucapollani** [comment]
I've stumbled on the same thing, I'm currently trying to reset row selection any time the data updates. I don't know whether it's the correct approach.

```js
	useEffect(() => {
		table.resetRowSelection(true);
	}, [data]);
```

Have a look also to this if interested: #4498

**Reply by dufia** [reply]
I ended up with a Map of selected rows (IDs) and custom logic in `onRowSelectionChange`

**Reply by scottdickerson** [reply]
@dufia do you have an example of the custom logic in `onRowSelectionChange`?   We're overriding the `getRowId` table option to set custom row ids so the key in the `onRowSelectionChange` is not the index of the row but a field of the datum in that row.  Since we don't have all the data in memory, we can't tweak the selectedRowState to select all of them when the Select All checkbox  is clicked

**Reply by dufia** [reply]
@scottdickerson  I deleted it since, my issue was solved by simply using `getRowId`

**Reply by scottdickerson** [reply]
thanks for answering @dufia
